core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -346,6 +348,112 @@ public void awaitTxn(String tableName, long txn, long timeout, TimeUnit timeoutU
+                                path.trimTo(pathLen).concat(tableToken.getDirName()).concat(MatViewRefreshState.MAT_VIEW_STATE_FILE_NAME);
+                                if (isMatViewStateExists) {
+                                    reader.of(path.$());
+                                    MatViewRefreshState.readFrom(reader, state);
+                                } else {
+                                    blockFileWriter.of(path.$());
+                                    MatViewRefreshState.append(state, blockFileWriter);
+                                }
+                            }
+                        } else {
+                            // Mat view state already exists, just load it
+                            path.trimTo(pathLen).concat(tableToken.getDirName()).concat(MatViewRefreshState.MAT_VIEW_STATE_FILE_NAME);
+                            reader.of(path.$());
+                            MatViewRefreshState.readFrom(reader, state);
+                        }
+
+                        if (state != null && !state.isInvalid()) {
+                            MatViewDefinition matViewDefinition = state.getViewDefinition();
+                            final TableToken baseTableToken = tableNameRegistry.getTableToken(matViewDefinition.getBaseTableName());
+                            final boolean baseTableExists = baseTableToken != null && !tableNameRegistry.isTableDropped(baseTableToken);
+                            if (!baseTableExists) {
+                                // Print a warning, but let the mat view load in invalid state.
+                                LOG.info().$("base table for materialized view does not exist [table=").utf8(matViewDefinition.getBaseTableName())
+                                        .$(", view=").utf8(tableToken.getTableName())
+                                        .I$();
+                            }
+                            // this should be called only during replica to primary transition
+                            if (checkMatViewConsistency) {
+
+                                long matViewLastTxn = getTableSequencerAPI().lastTxn(tableToken);
+                                long baseTableLastTxn = baseTableExists && baseTableToken.isWal()
+                                        ? getTableSequencerAPI().lastTxn(baseTableToken)
+                                        : Long.MAX_VALUE;
+                                if (state.getSeqTxn() > matViewLastTxn || state.getLastRefreshBaseTxn() > baseTableLastTxn) {
+                                    // Materialized view refresh state is ahead of the base table or the view itself.
+                                    // This may happen when a read-only replica node which isn't fully caught up
+                                    // was converted to primary.
+                                    try (TransactionLogCursor transactionLogCursor = getTableSequencerAPI().getCursor(tableToken, matViewLastTxn - 1)) {
```

Comment:
This can throw, please handle the exception

Reply:
fixed